flow/size.yaml

```diff
@@ -62,3 +67,70 @@ components:
           type: integer
           default: 1518
           x-field-uid: 2
+    Flow.SizeWeightPairs:
+      type: object
+      description: |-
+        Frame size distribution, defined as <size, weight> pairs (including IMIX distribution).
+        Frames are randomly generated such that the proportion of each frame size out of the total number of frames 
+        are matching with the weight value of the <size, weight> pair. However, as with any other probability 
+        distribution, the sample distribution is close to theoretical value only if the size of the sample is reasonably large. 
+        When the number of frames is very low the transmitted frames may not come close to the ratio described in the weight.
+      properties:
+        choice:
+          type: string
+          default: predefined
+          x-field-uid: 1
+          x-enum:
+            predefined:
+              x-field-uid: 1
+            custom:
+              x-field-uid: 2
+        predefined:
+          type: string
+          default: imix
+          description: |-
+            Specify predefined frame size distribution <size, weight> pairs (including IMIX distribution).  
+            The available predefined distribution pairs are:
+            - IMIX (64:7, 570:4, and 1518:1)  
+            - IPSec IMIX (90:58.67, 92:2, 594:23.66 and 1418:15.67)  
+            - IPv6 IMIX (60:58.67, 496:2, 594:23.66 and 1518:15.67)  
+            - Standard IMIX (58:58.67, 62:2, 594:23.66 and 1518:15.67)  
+            - TCP IMIX (90:58.67, 92:2, 594:23.66 and 1518:15.67)  
+          x-field-uid: 2
+          x-enum:
+            imix:
+              x-field-uid: 1
+            ipsec_imix:
+              x-field-uid: 2
+            ipv6_imix:
+              x-field-uid: 3
+            standard_imix:
+              x-field-uid: 4
+            tcp_imix:
+              x-field-uid: 5
+        custom:
+          type: array
+          items:
+            $ref: '#/components/schemas/Flow.SizeWeightPairs.Custom'
+          x-field-uid: 3
+    Flow.SizeWeightPairs.Custom:
+      type: object
+      description: |-
+        Custom frame size distribution <size, weight> pair.
+      properties:
+        size:
+          description: The size of the frame (in bytes) for this weight pair.
+          type: integer
+          format: uint32
+          minimum: 12
+          maximum: 65535
+          default: 64
+          x-field-uid: 1
+        weight:
+          description: |-
+            Weight assigned to the corresponding frame size in this weight pair. 
```

Comment:
I am assuming that all the weights have to add up to 100?  If so, we should clearly call it out and also define the behavior when the weights are <100 or >100.  Or if it is implementation specific/un-defined, then explicitly state that.

Reply:
It is not necessary that all the weights have to add up to 100. For example, we may have weight pairs defined as (64,1) and (1500, 1000). It means, for every 1001 frames transmitted, 1 packet of 64 bytes, and the remaining 1000 packets of 1500 bytes.